lib/src/model.dart

```diff
@@ -1299,6 +1309,25 @@ class Library extends ModelElement {
         new NamespaceBuilder().createExportNamespaceForLibrary(element);
   }
 
+  List<String> _allOriginalModelElementNames;
+
+  /// [allModelElements] resolved to their original names.
+  ///
+  /// A collection of [ModelElement.fullyQualifiedNames] for [ModelElement]s
+  /// documented with this library, but these ModelElements and names correspond
+  /// to the defining library where each originally came from with respect
+  /// to inheritance and reexporting.  Most useful for error reporting.
+  Iterable<String> get allOriginalModelElementNames {
```

Comment:
Curious why you're not returning a `List` here.  Is this to prevent write access?